crates/matrix-sdk-crypto/src/machine.rs

```diff
@@ -1126,6 +1126,22 @@ impl OlmMachine {
         &self,
         sync_changes: EncryptionSyncChanges<'_>,
     ) -> OlmResult<(Vec<Raw<AnyToDeviceEvent>>, Vec<RoomKeyInfo>)> {
+        let (events, changes) = self.receive_sync_changes_helper(sync_changes).await?;
+
+        // Technically save_changes also does the same work, so if it's slow we could
+        // refactor this to do it only once.
+        let room_key_updates: Vec<_> =
+            changes.inbound_group_sessions.iter().map(RoomKeyInfo::from).collect();
+
+        self.store().save_changes(changes).await?;
+
+        Ok((events, room_key_updates))
+    }
+
+    pub(crate) async fn receive_sync_changes_helper(
```

Comment:
I don't like `_helper` functions, the name doesn't really say anything about what it does or how it relates to the non-`_helper` function. Since it seems to be called *at the start* of `receive_sync_changes`, would `preprocess_sync_changes` be a reasonable name?

Reply:
Renamed in 404ce5698b237ec2280f72718e0a1a7a89773396.